tests/streams/bstream.h

```diff
@@ -253,9 +253,17 @@ template <class TStreamIn, class TStreamOut> class BaseStreamTestCase : public C
         while (stream_in.IsOk())
         {
             char peekChar = stream_in.Peek();
+            size_t peekLastRead = stream_in.LastRead();
```

Comment:
I want to highlight this test update as something I'm not entirely sure about, since I believe it would impose this behavior of `Peek` on all stream types. This seems to be true at the moment, though as PB mentioned on the issue, this behavior is not formally documented.

It is mentioned by the in-source comments on `wxInputStream::LastRead` that `Peek` affects it, and on `wxInputStream::Peek` that the last read count is set to 0 or 1 though, which is why I thought to update the common tests here instead of just adding one against the memory stream specifically.

I'm wondering if I should change this test to just be against the memory stream, or perhaps this change might warrant an update to the doxygen comments for `wxInputStream::Peek` and/or `LastRead` as well?

Reply:
It does make sense to impose the same behaviour for all stream types, however, and as all the existing tests pass, it seems that this is already the case, so I think it's fine to do this.